### PR TITLE
fix operator precedence

### PR DESCRIPTION
**Pull request by autoboosh**
The operator precedence was a total chaos for some reason. It also seems like there's a mixup between statement and expression here since about half of the rule could/should be moved to some sort of expr-rule. I found this paper describing how ANTLR4 works with left-recursion very useful: [Adaptive LL(*) Parsing: The Power of Dynamic Analysis](http://www.antlr.org/papers/allstar-techreport.pdf). Appendix C is where left-recursion and how the precedence rules are implemented.
VBA operator precedence rules are described here: [5.6.9.1 Operator Precedence and Associativity](https://msdn.microsoft.com/en-us/library/ee156871.aspx)

Removed the `PLUS valueStmt` rule as well since that gets "optimized away" as soon as you move to a new line in the VBE. It's not mentioned in the VBA specification either.

PLUS_EQ and MINUS_EQ are gone now too since those are not supported in VBA. I don't quite understand this one to be honest. Is there some kind of out-of-spec VBA runtime out there supporting these operators?

### CODE REVIEW

**Comment by retailcoder** [comment]
I don't know what += and -= operators were doing there either. Saw them before, but they didn't interfere with anything and since we didn't need to care about those... I just left them there. Nice cleanup!

File: Rubberduck.Parsing/Grammar/VBA.g4 [hunk 1]
```diff
@@ -74,7 +74,9 @@
 *   - implemented proper specs for DATELITERAL.
 *   - added comments to parse tree (removes known limitation #2).
 *   - macroConstStmt now allowed in blockStmt.
-*   - allow type hints for parameters.
```
Comment (retailcoder) [inline Rubberduck.Parsing/Grammar/VBA.g4]: Accidental deletion?

File: Rubberduck.Parsing/Grammar/VBA.g4 [hunk 2]
```diff
@@ -900,7 +889,8 @@ XOR : X O R;
 // symbols
 AMPERSAND : '&';
 ASSIGN : ':=';
-DIV : '\\' | '/';
+DIV : '/';
+INTDIV : '\\';
```
Comment (retailcoder) [inline Rubberduck.Parsing/Grammar/VBA.g4]: Ooh, nice catch! :+1:

File: Rubberduck.Parsing/Grammar/VBA.g4 (hunk 1 above)
Comment (autoboosh) [inline Rubberduck.Parsing/Grammar/VBA.g4]: Yes, my bad. Why are we keeping a manual changelog btw? Is this for merging with the ANTLR4 grammar repository?

Comment (retailcoder) [inline Rubberduck.Parsing/Grammar/VBA.g4]: Yeah, well... it was there (up to v1.2) when I grabbed the grammar file.. I just kept it going.

**Pull request merged**